mountpoint-s3/src/prefetch.rs

```diff
@@ -199,6 +214,9 @@ where
     }
 }
 
+type ParsedMetadata = Arc<RwLock<Option<HashMap<(usize, usize), (u64, u64)>>>>;
+type RawMetadata = Arc<RwLock<Option<Bytes>>>;
```

Comment:
Nice, I like what you did here with setting up new type aliases.

I'd argue that the most benefit we could get is aliasing the key and value in the HashMap (although the overall aliases are still good for brevity).

Something like this would be neat:

```suggestion
/// Mapping of row groups and columns to their start and end offsets.
type ParsedMetadata = Arc<RwLock<Option<HashMap<(ColumnIndex, RowGroupIndex), (u64, u64)>>>>;
type RawMetadata = Arc<RwLock<Option<Bytes>>>;

/// Index of the row group within the file.
type RowGroupIndex = usize;
/// Index of the column within a row group.
type ColumnIndex = usize;
```

Even better, if you use those aliases later then the type system can fail compilation if you ever swap column and row groups!

```rust
let mut mapping: HashMap<(RowGroupIndex, ColumnIndex), (u64, u64)> = todo!("hashmap from somewhere");

let rowGroupIdx: RowGroupIndex = 12;
let colIdx: ColumnIndex = 5;

// I think this will fail to compile. You could try it! :)
mapping.insert((colIdx, rowGroupIdx), (1024, 2048));
```

Reply:
Actually, that last bit isn't as feasible as I thought. I should have checked it before posting. 😄 

It's possible, but we'd probably we'd need to use new 1-tuple struct instead. Don't worry about that for now.

Reply:
Yeah, this is a really good suggestion, I'll add that!